src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,51 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int KEY_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private final Object[][] objects;
+    private int positionIndex;
+
+    public StorageImpl() {
+        this.objects = new Object[2][10];
+        this.positionIndex = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = this.getKeyIndex(key);
+        if (index == -1) {
+            objects[KEY_INDEX][positionIndex] = key;
+            objects[VALUE_INDEX][positionIndex] = value;
+            positionIndex++;
+        } else {
+            objects[VALUE_INDEX][index] = value;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int keyIndex = this.getKeyIndex(key);
+        return keyIndex != -1 ? (V) objects[VALUE_INDEX][keyIndex] : null;
     }
 
     @Override
     public int size() {
+        return positionIndex;
+    }
+
+    private int getKeyIndex(K key) {
+        for (int i = 0; i < positionIndex; i++) {
+            if (key == null) {
+                if (objects[KEY_INDEX][i] == null) {
+                    return i;
+                }
+            } else {
+                if (key.equals(objects[KEY_INDEX][i])) {
+                    return i;
+                }
+            }
+        }
```

Comment:
```suggestion
        for (int i = 0; i < positionIndex; i++) {
            if (key == null) {
                if (objects[KEY_INDEX][i] == key || key != null && key.equals(objects[KEY_INDEX][i])) {
                    return i;
                }
            }
        }
```
should work perfectly

Reply:
Idea advised "Condition 'key != null && key.equals(objects[KEY_INDEX][i])' is always 'false' when reached "
because we enter if only if key == null, so how it then could be != null